### PR TITLE
Implement new invariant for dev and acc fees with scheduled execution

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Implemented a new invariant for developers and accumulated fees, taking into account also the scheduled transactions executed
* Refactored gasConsumed in gasProvided

### CODE REVIEW

**Review by AdoAdoAdo — CHANGES_REQUESTED** [review]

File: process/block/baseProcess.go [hunk 1]
```diff
@@ -1613,56 +1606,50 @@ func (bp *baseProcessor) getFeesAndGasMetrics() scheduled.GasAndFees {
...
 
-	deltaAccumulatedFees := big.NewInt(0).Sub(finalMetrics.AccumulatedFees, initialMetrics.AccumulatedFees)
+	deltaAccumulatedFees := big.NewInt(0).Sub(finalGasAndFees.AccumulatedFees, initialGasAndFees.AccumulatedFees)
 	if deltaAccumulatedFees.Cmp(zero) < 0 {
-		log.Error("feeAndGasMetricsDelta",
-			"initial accumulatedFees", initialMetrics.AccumulatedFees.String(),
-			"final accumulatedFees", finalMetrics.AccumulatedFees.String(),
+		log.Error("gasAndFeesDelta",
+			"initial accumulatedFees", initialGasAndFees.AccumulatedFees.String(),
+			"final accumulatedFees", initialGasAndFees.AccumulatedFees.String(),
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/baseProcess.go:1617]: done

File: process/block/baseProcess.go [hunk 2]
```diff
@@ -1613,56 +1606,50 @@ func (bp *baseProcessor) getFeesAndGasMetrics() scheduled.GasAndFees {
...
 
-	deltaDevFees := big.NewInt(0).Sub(finalMetrics.DeveloperFees, initialMetrics.DeveloperFees)
+	deltaDevFees := big.NewInt(0).Sub(initialGasAndFees.DeveloperFees, initialGasAndFees.DeveloperFees)
 	if deltaDevFees.Cmp(zero) < 0 {
-		log.Error("feeAndGasMetricsDelta",
-			"initial devFees", initialMetrics.DeveloperFees.String(),
-			"final devFees", finalMetrics.DeveloperFees.String(),
+		log.Error("gasAndFeesDelta",
+			"initial devFees", initialGasAndFees.DeveloperFees.String(),
+			"final devFees", initialGasAndFees.DeveloperFees.String(),
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/baseProcess.go:1626]: done

File: process/block/baseProcess.go [hunk 3]
```diff
@@ -1613,56 +1606,50 @@ func (bp *baseProcessor) getFeesAndGasMetrics() scheduled.GasAndFees {
...
 
-	deltaGasProvided := int64(finalMetrics.GasProvided) - int64(initialMetrics.GasProvided)
+	deltaGasProvided := int64(initialGasAndFees.GasProvided) - int64(initialGasAndFees.GasProvided)
 	if deltaGasProvided < 0 {
-		log.Error("feeAndGasMetricsDelta",
-			"initial gasProvided", initialMetrics.GasProvided,
-			"final gasProvided", finalMetrics.GasProvided,
+		log.Error("gasAndFeesDelta",
+			"initial gasProvided", initialGasAndFees.GasProvided,
+			"final gasProvided", initialGasAndFees.GasProvided,
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/baseProcess.go:1635]: done

File: process/block/baseProcess.go [hunk 4]
```diff
@@ -1613,56 +1606,50 @@ func (bp *baseProcessor) getFeesAndGasMetrics() scheduled.GasAndFees {
...
-			"final accumulatedFees", finalMetrics.AccumulatedFees.String(),
+		log.Error("gasAndFeesDelta",
+			"initial accumulatedFees", initialGasAndFees.AccumulatedFees.String(),
+			"final accumulatedFees", initialGasAndFees.AccumulatedFees.String(),
 			"error", process.ErrNegativeValue)
 		return result
 	}
 
-	deltaDevFees := big.NewInt(0).Sub(finalMetrics.DeveloperFees, initialMetrics.DeveloperFees)
+	deltaDevFees := big.NewInt(0).Sub(initialGasAndFees.DeveloperFees, initialGasAndFees.DeveloperFees)
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/baseProcess.go:1622]: :+1: done

File: process/block/baseProcess.go [hunk 5]
```diff
@@ -1613,56 +1606,50 @@ func (bp *baseProcessor) getFeesAndGasMetrics() scheduled.GasAndFees {
...
-			"final devFees", finalMetrics.DeveloperFees.String(),
+		log.Error("gasAndFeesDelta",
+			"initial devFees", initialGasAndFees.DeveloperFees.String(),
+			"final devFees", initialGasAndFees.DeveloperFees.String(),
 			"error", process.ErrNegativeValue)
 		return result
 	}
 
-	deltaGasProvided := int64(finalMetrics.GasProvided) - int64(initialMetrics.GasProvided)
+	deltaGasProvided := int64(initialGasAndFees.GasProvided) - int64(initialGasAndFees.GasProvided)
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/baseProcess.go:1631]: :+1: done

File: process/block/baseProcess.go [hunk 6]
```diff
@@ -1613,56 +1606,50 @@ func (bp *baseProcessor) getFeesAndGasMetrics() scheduled.GasAndFees {
...
-			"final gasProvided", finalMetrics.GasProvided,
+		log.Error("gasAndFeesDelta",
+			"initial gasProvided", initialGasAndFees.GasProvided,
+			"final gasProvided", initialGasAndFees.GasProvided,
 			"error", process.ErrNegativeValue)
 		return result
 	}
 
-	deltaGasPenalized := int64(finalMetrics.GasPenalized) - int64(initialMetrics.GasPenalized)
+	deltaGasPenalized := int64(initialGasAndFees.GasPenalized) - int64(initialGasAndFees.GasPenalized)
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/baseProcess.go:1640]: :+1: done

File: process/block/baseProcess.go [hunk 7]
```diff
@@ -1613,56 +1606,50 @@ func (bp *baseProcessor) getFeesAndGasMetrics() scheduled.GasAndFees {
...
 	if deltaGasPenalized < 0 {
-		log.Error("feeAndGasMetricsDelta",
-			"initial gasPenalized", initialMetrics.GasPenalized,
-			"final gasPenalized", finalMetrics.GasPenalized,
+		log.Error("gasAndFeesDelta",
+			"initial gasPenalized", initialGasAndFees.GasPenalized,
+			"final gasPenalized", initialGasAndFees.GasPenalized,
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/baseProcess.go:1644]: done

File: process/block/baseProcess.go [hunk 8]
```diff
@@ -1613,56 +1606,50 @@ func (bp *baseProcessor) getFeesAndGasMetrics() scheduled.GasAndFees {
...
 			"error", process.ErrNegativeValue)
 		return result
 	}
-	deltaGasRefunded := int64(finalMetrics.GasRefunded) - int64(initialMetrics.GasRefunded)
+	deltaGasRefunded := int64(initialGasAndFees.GasRefunded) - int64(initialGasAndFees.GasRefunded)
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/baseProcess.go:1648]: :+1: done

File: process/block/baseProcess.go [hunk 9]
```diff
@@ -1613,56 +1606,50 @@ func (bp *baseProcessor) getFeesAndGasMetrics() scheduled.GasAndFees {
...
 	if deltaGasRefunded < 0 {
-		log.Error("feeAndGasMetricsDelta",
-			"initial gasRefunded", initialMetrics.GasRefunded,
-			"final GasRefunded", finalMetrics.GasRefunded,
+		log.Error("gasAndFeesDelta",
+			"initial gasRefunded", initialGasAndFees.GasRefunded,
+			"final gasRefunded", initialGasAndFees.GasRefunded,
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/baseProcess.go:1652]: done

File: process/block/preprocess/basePreProcess.go [hunk 10]
```diff
@@ -384,14 +384,14 @@ func (bpp *basePreProcess) requestMissingTxsForShard(
 	return requestedTxs
 }
 
-func (bpp *basePreProcess) computeGasConsumedByTx(
+func (bpp *basePreProcess) computeGasProvidedByTx(
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/preprocess/basePreProcess.go:387]: done

**Review by AdoAdoAdo — APPROVED** [review]

File: process/coordinator/process.go [hunk 11]
```diff
@@ -3,6 +3,7 @@ package coordinator
 import (
 	"bytes"
 	"fmt"
+	"github.com/ElrondNetwork/elrond-go-core/core/atomic"
```
Comment (raduchis) [inline process/coordinator/process.go:6]: imports are not sorted

**Review by raduchis — APPROVED** [review]

**Pull request merged**